extensions/amp-auto-lightbox/0.1/carousel-criteria.js

```diff
@@ -0,0 +1,294 @@
+      const actions = Services.actionServiceForDoc(element);
+      return tryResolve(() =>
+        !actions.hasResolvableAction(element, 'tap', actionLookupStopAt));
+    }
+
+    const img = element.querySelector('amp-img');
+    if (!img) {
+      return resolveFalse();
+    }
+
+    const slideMeetsSizingPromise =
+        SlideCriteria.meetsSizingCriteria(img, element);
+
+    return slideMeetsSizingPromise.then(slideMeetsSizingCriteria => {
+      if (!slideMeetsSizingCriteria) {
+        return false;
+      }
+      return SlideCriteria.meetsAllSync(element, /* depth */ 0);
+    });
+  }
+
+  /**
+   * @param {!Element} element
+   * @param {number} depth
+   * @return {boolean}
+   */
+  static meetsAllSync(element, depth) {
+    if (depth > MAX_TRAVERSING_DEPTH) {
+      return false;
+    }
+    if (!isElementAdmissible(element)) {
+      return false;
+    }
+    // Already traversed up the tree, we only need to check the immediate
+    // action.
+    const actionLookupStopAt = dev().assertElement(element.parentElement);
+    const actions = Services.actionServiceForDoc(element);
+    if (actions.hasResolvableAction(element, 'tap', actionLookupStopAt)) {
+      return false;
+    }
+    if (!shouldElementBeTraversed(element)) {
+      return true;
+    }
+    const {children} = element;
+    if (children.length > MAX_TRAVERSING_BREADTH) {
+      return false;
+    }
+    return toArray(children).every(child =>
+      SlideCriteria.meetsAllSync(child, depth + 1));
+  }
+
+  /**
+   * @param {!AmpElement} img
+   * @param {!Element} slide
+   * @return {!Promise<boolean>}
+   */
+  static meetsSizingCriteria(img, slide) {
+    devAssert(img.tagName == 'AMP-IMG');
+
+    return img.getImpl().then(impl => new Promise(resolve => {
+      impl.measureElement(() => {
+        const {
+          width: imgWidth,
+          height: imgHeight,
+        } = img.getLayoutBox();
+
+        const {
+          width: slideWidth,
+          height: slideHeight,
+        } = slide./*OK*/getBoundingClientRect();
+
+        const imgArea = imgWidth * imgHeight;
+        const slideArea = slideWidth * slideHeight;
+
+        resolve((imgArea / slideArea) >= MIN_IMG_SLIDE_AREA_RATIO);
+      });
+    }));
+  }
+}
+
+/**
+ * @param {!Element} element
+ * @return {boolean}
+ */
+function isElementAdmissible(element) {
+  if (isAmpElement(element)) {
+    return isAmpElementAdmissible(element);
+  }
+  const {tagName} = element;
+  if (ADMISSIBLE_HTML_ELEMENTS[tagName]) {
+    return (tagName != 'A' || !element.hasAttribute('href'));
+  }
+  return false;
+}
+
+/**
+ * @param {!Element} element
+ * @return {boolean}
+ */
+function isAmpElementAdmissible(element) {
+  const {tagName} = element;
+  if (ADMISSIBLE_AMP_ELEMENTS[tagName]) {
+    return true;
+  }
+  if (element.getAttribute('layout') == Layout.NODISPLAY) {
+    return (tagName.indexOf('LIGHTBOX') < 0);
+  }
+  if (element.children.length == 1 &&
```

Comment:
Is there any documentation that could explain why an AMP element with a `<script>` as its first child is admissible?